src/controllers/dynakube/version/updater.go

```diff
@@ -180,15 +189,21 @@ func updateVersionStatusForTenantRegistry( //nolint:revive
 }
 
 func getTagFromImageID(imageID string) (string, error) {
-	ref, err := reference.Parse(imageID)
+	ref, err := name.ParseReference(imageID)
 	if err != nil {
 		return "", err
 	}
-	taggedRef, ok := ref.(reference.NamedTagged)
-	if !ok {
-		return "", errors.New("no tag found to check for downgrade")
+
+	if taggedRef, ok := ref.(name.Tag); ok {
+		return taggedRef.TagStr(), nil
+	} else if digestRef, ok := ref.(name.Digest); ok {
+		taggedStr := strings.TrimSuffix(digestRef.String(), digestDelimiter+digestRef.DigestStr())
+
+		if taggedRef, err = name.NewTag(taggedStr); err == nil {
+			return taggedRef.TagStr(), nil
+		}
 	}
```

Comment:
what are we doing here ? and why? (we didn't use to care about digest here)

Reply:
Removing the digest from input strings like `my.reg/rep:1.2.3@sha256:abcdcoffee`

Reply:
so in case `my.reg/rep:1.2.3@sha256:abcdcoffee` then  `_, ok := ref.(name.Tag)` results in `ok == false`?
if that is the case, then I am not angry, just disappointed in the library 😓 

otherwise it does not make sense

Reply:
just debuged the tests, and yes it does that 😭

Reply:
also during the debugging the tests (nice tests btw ⭐ ), in the case of `some.registry.com@sha256:7ece13a07a20c77a31cc36906a10ebc90bd47970905ee61e8ed491b7f4c5d62f` (so when only digest given), then we should return the error, as we don't know the tag. It shouldn't return `latest`, as we couldn't know if a given digest IS or not latest.

we only use this function when dealing with the PublicRegsitry feature, where we will always get a tag and not a digest, so we should error if that behaviour changes.

Reply:
That's another caveat of the new lib, that it adds `latest` as a default fallback without asking. I'll have a closer look how to make this happen.

Reply:
Fixed this and updated unit tests accordingly.